core/src/main/java/org/apache/brooklyn/enricher/stock/Enrichers.java

```diff
@@ -676,6 +681,62 @@ public String toString() {
                     .toString();
         }
     }
+
+    protected abstract static class AbstractReducerBuilder<S, T, B extends AbstractReducerBuilder<S, T, B>> extends AbstractEnricherBuilder<B> {
+        protected AttributeSensor<T> publishing;
+        protected Entity fromEntity;
+        protected List<AttributeSensor<S>> reducing;
+        protected Function<List<S>, T> computing;
+        protected String functionName;
+        private Map<String, Object> parameters;
+
+        public AbstractReducerBuilder(Class<? extends Reducer<S, T>> clazz, List<AttributeSensor<S>> val) {
```

Comment:
This would be inconsistent with the other uses throughout, and could cause confusion.  I'm inclined to say that this should be done in another PR.  Moreover, this is a protected class and not public.  Perhaps a discussion on the mailing list about whether this should be adopted widely.